frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/import-feature-flag-modal/import-feature-flag-modal.component.ts

```diff
@@ -0,0 +1,93 @@
+import { ChangeDetectionStrategy, Component, ElementRef, Inject, OnInit, ViewChild } from '@angular/core';
+import { CommonModalComponent } from '../../../../../shared-standalone-component-lib/components';
+import { MAT_DIALOG_DATA, MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { CommonModalConfig } from '../../../../../shared-standalone-component-lib/components/common-modal/common-modal-config';
+import { FeatureFlagsService } from '../../../../../core/feature-flags/feature-flags.service';
+import { BehaviorSubject, Observable, Subscription, combineLatest, map } from 'rxjs';
+import { CommonModule } from '@angular/common';
+import { SharedModule } from '../../../../../shared/shared.module';
+
+@Component({
+  selector: 'app-import-feature-flag-modal',
+  standalone: true,
+  imports: [CommonModalComponent, CommonModule, SharedModule],
+  templateUrl: './import-feature-flag-modal.component.html',
+  styleUrl: './import-feature-flag-modal.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ImportFeatureFlagModalComponent {
+  @ViewChild('fileInput') fileInput: ElementRef<HTMLInputElement>;
+
+  isImportActionBtnDisabled = new BehaviorSubject<boolean>(true);
+  isDragOver = false;
```

Comment:
let's make `isDragOver` and `fileName` BehaviorSubjects as well please to be consistently preferring Observable streams in our new components. You'd use `async` pipe in the html.